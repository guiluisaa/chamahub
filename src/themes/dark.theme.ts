import { DefaultTheme } from 'styled-components';

import { borderRadius } from './tokens/border.token';
import breakpoint from './tokens/breakpoint.token';
import { darkThemeButton } from './tokens/button.token';
import color from './tokens/color.token';
import { fontWeight } from './tokens/font.token';
import { darkThemeInput } from './tokens/input.token';

const darkTheme: DefaultTheme = {
  theme: 'dark',
  color,

  font: {
    family: "'Inter', sans-serif",
    weight: fontWeight,
    color: {
      primary: color.white,
      secondary: color.deepGray,
    },
  },

  border: {
    radius: borderRadius,
    color: '#333',
  },

  breakpoint,

  background: {
    color: {
      primary: color.black,
      secondary: color.deepBlack,
    },
  },

  components: {
    button: darkThemeButton,
    input: darkThemeInput,
  },
};

export default darkTheme;
