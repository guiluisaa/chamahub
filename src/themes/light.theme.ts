import { DefaultTheme } from 'styled-components';

import { borderRadius } from './tokens/border.token';
import breakpoint from './tokens/breakpoint.token';
import color from './tokens/color.token';
import { fontWeight } from './tokens/font.token';

const lightTheme: DefaultTheme = {
  theme: 'light',
  color,

  font: {
    family: "'Inter', sans-serif",
    weight: fontWeight,
    color: {
      primary: color.black,
      secondary: color.gray,
    },
  },

  border: {
    radius: borderRadius,
    color: color.gray,
  },

  breakpoint,

  background: {
    color: {
      primary: '#f9f9f9',
      secondary: color.white,
    },
  },
};

export default lightTheme;
