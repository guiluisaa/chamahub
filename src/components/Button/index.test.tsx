import React from 'react';
import { shallow } from 'enzyme';

import Button from '@/components/Button';
import { lightTheme } from '@/themes';

describe('<Button />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<Button theme={lightTheme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
