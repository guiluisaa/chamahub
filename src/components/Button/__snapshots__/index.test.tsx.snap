// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapchat 1`] = `
<button
  className="sc-bczRLJ dtwVFu"
/>
`;
